client/tx/tx.go

```diff
@@ -322,6 +322,7 @@ func PrepareFactory(clientCtx client.Context, txf Factory) (Factory, error) {
 func SignWithPrivKey(
 	signMode signing.SignMode, signerData authsigning.SignerData,
 	txBuilder client.TxBuilder, priv crypto.PrivKey, txConfig client.TxConfig,
+	accSeq uint64,
```

Comment:
How did this function work before without that accSeq?

`txConfig.SignModeHandler().GetSignBytes(signMode, signerData, txBuilder.GetTx())` must have access to the account sequence somehow in order to create the proper sign bytes.

Reply:
Yes, [`signerData.AccountSequence`](https://github.com/cosmos/cosmos-sdk/blob/60f7edfe3dbfefecd8e2ee5fc24c5bebc4461c11/x/auth/signing/sign_mode_handler.go#L23-L35) contains the same info.

Much better to remove this second argument that must be the same, and just reuse the number from signerData.

This would also reduce some need for plumbing

Reply:
I thought about this, but decided not to do that.

The reason is, in SIGN_MODE_DIRECT, putting Sequence in SignerData is confusing because it won't be used. I put a comment on SignerData, to warn that Sequence is only used in SIGN_MODE_LEGACY.

I see SIGN_MODE_DIRECT as the default/preferred mode, in which case `SignWithPrivKey` would have exactly this signature above. I'm just leaving Sequence in SignerData for legacy reasons.

Reply:
Fair enough. I do believe that field will be there for > 1 year, probably forever, as all exchanges will have to upgrade their hardware signers to use the DIRECT format before we can consider removing legacy mode.

It is simpler to just use it, but not wrong the way you did. I'll leave this to you all to decide.

Reply:
Right, I also think that field will be there for a long time.

I would advocate for explicitness over simplicity here. With this signature, a JS dev looking for the reference implementation of DIRECT in this repo would not need to know that the sequence is actually hidden in SignerData in our implementation.

Yes, I'm also happy to hear other SDK devs opinions.

Reply:
@ethanfrey Would you be okay to merge like this? This is a SDK-internal detail, we can find a better solution later, if needed.

Reply:
Yes, I am fine to merge as is. It is internals, but some internals we stumble upon sometimes.